integrated_channels/integrated_channel/exporters/learner_data.py

```diff
@@ -175,6 +173,7 @@ def single_assessment_level_export(self, **kwargs):
         if not (TransmissionAudit and already_transmitted) and LearnerExporter.has_data_sharing_consent(
                 enterprise_enrollment):
 
+            course_details = None
```

Comment:
I think the loss of caching might affect something, but Alex will have more insight into that bit. 
For this particular method though, I have to ask: why are we even retrieving course_details? We seem to be using it for an if condition, but I feel like there must be a way to do that without another API call as we aren't actually using any of the contents in this method and we're not setting it into the class variable so it can be reused without another API call now, so I'm just unclear why the heavy lift of another API call here.

I also know this PR doesn't address that, so it's something maybe Alex can explain or we can address at another time.

Reply:
good point!

Reply:
yeah I too not clear now that you bring it up, why we fetch it here, thoughts @alex-sheehan-edx ?

Reply:
I see it's used as a check right after, that is the only use of it in this function. also get_course_Details is no longer a REST api call so that should help us avoid perf issues

Reply:
also course_details would be per course_run_id, so if we cached, we would have to do per course id or course_run_id anyway

Reply:
In terms of caching, this is a single enrollment transmission, so I don't think it would ever actually have an effect (?). Meaning that we would cache bulk transmissions by retaining the fetched course details to be reused if applicable, but since we're only transmitting for one learner in one course, I don't think it would have an effect.

(I also took a look at the commit when I had originally added this method and it included this:
https://github.com/edx/edx-enterprise/commit/ca1750a93b187f3a9e998965269fbcfb4212f77b#diff-f488bfc03a35af19cc710b17a06ab3014e240abef4ad52114822c8b0e081b5caR177-R181 )
 
Having said that... As to why we're even fetching the course details here... I'm really not entirely sure 🙃. The assessment level exports (bulk and single) were largely based on the (not so perfect) completion data export method, which uses the course details to:
- confirm that the course exists before attempting to export
- uses the `pacing` attribute of `course_details` to determine whether it should pull certificate data or grades data
- uses the `end` attribute in retrieving grades data (for self paced courses)
- some light logging, giving details as to the specific transmission happening

This `single_assessment_level_export` does not need:
- to differentiate between pacings, we simple grab the assessment grade data 
- to use `end` as these transmissions happen throughout the learners progress through the course
- any kind of caching (that I can tell?) because we are only transmitting data related to a single course (we will only have one course_details throughout the entire transmission)

So the only thing, as you all have rightly pointed out, that it's using the course details for is to verify that the course exists. However having said that, if we take a look at `bulk_assessment_level_export` above this method, we are accomplishing essentially the same goal but for many learners and we never fetch course details a single time. 

So TL:DR I think fetching the course details was a consequence of building off of existing, not perfect architecture and can probably be removed entirely from this method, if we're ok with not validating that the course exists.

Reply:
Is there a scenario where we'd get data to export for a course that does not in fact exist?
I would imagine we'd not get to this point if we just didn't have a course? Or we could also just verify the course details where we actually use them.

Reply:
We start the whole export when a user has gone onto our site and physically completed the course, so I really don't see how that could ever happen without the course existing.

Reply:
In fact, we use the `course run id` provided by platform's signal to get those course details so if we're worried about not having a valid course, I would ask how the learner was able to complete a subsection of the course linked to the course run id provided by the signal (which directly pulls it from the course)

Reply:
I agree there too, why fetch again if we are here as a result of a course's completion.. right?

Reply:
going to bite the bullet and try removing it ...

Reply:
but this also. makes me ask why this?

```
       course_run_id = kwargs.get('course_run_id', None)
```

seems like course_run_id should then be a required arg , do we want to at least check course_run_id is not None here before proceeding?

Reply:
or we can just let the enrollment_queryset be empty if it's None, don't know if that is going to be the case

Reply:
I would say let's just not get course_details for now and see...

Reply:
okie check out the update!!! I still have not been able to test this locally (damn publisher/ecom etc!) so trying to run that task directly to see what happens ... with manually fed data

Reply:
`course_run_id = kwargs.get('course_run_id', None)` is another relic of the combined single and bulk completion exports. Most likely we can just change it to a required param.